klippy/extras/virtual_sdcard.py

```diff
@@ -57,6 +57,10 @@ def get_status(self, eventtime):
         if self.work_timer is not None and self.file_size:
             progress = float(self.file_position) / self.file_size
         return {'progress': progress}
+    def is_active(self):
+        if self.must_pause_work or self.work_timer is not None:
+            return True
```

Comment:
This looks a little odd to me - must_pause_work is a flag to indicate that the background work should be paused.  If a virtual sd card print is in progress than must_pause_work would always be false, but if a print was not in progress it could be either true or false.

Also, separately, the commit message for this commit says "Report current file position while paused when M27 is requested", but I don't see a change like that in the commit.

Reply:
When reviewing the code I had considered adding a new is_paused member, however it seemed to me that it would be set identically to must_pause_work.  It seems that must_pause_work is initialized to False, assigned to True in handle_shutdown and cmd_M25, and assigned to False in cmd_M24.  My conclusion that was that if must_pause_work is set to True then it must be paused unless it is shutdown.

It won't be a problem to add another member to track this though, in the event that must_pause_work is set to True when a print is not actually in progress.

Reply:
Ah - when I saw is_active() I thought it was semantically "is_actively_printing()".  If you're looking for an is_paused(), wouldn't that be `!is_actively_printing() and self.current_file is not None and self.file_position` ?

-Kevin

Reply:
Yes, I can see how it was confusing.  What I was looking for was to determine if the source of the gcode was the virtual_sdcard or octoprint, so I needed it to return true if it was printing or paused.

What I have decided to do was make is_active() semantically correct, it will return True if the sdcard is printing.